engine/access/rest/websockets/controller.go

```diff
@@ -44,12 +53,46 @@ func NewWebSocketController(
 
 // HandleConnection manages the WebSocket connection, adding context and error handling.
 func (c *Controller) HandleConnection(ctx context.Context) {
-	//TODO: configure the connection with ping-pong and deadlines
+	// configuring the connection with appropriate read/write deadlines and handlers.
+	err := c.configureConnection()
+	if err != nil {
+		// TODO: add error handling here
+		c.logger.Error().Err(err).Msg("error configuring connection")
+		c.shutdownConnection()
+		return
+	}
+
 	//TODO: spin up a response limit tracker routine
 	go c.readMessagesFromClient(ctx)
 	c.writeMessagesToClient(ctx)
 }
 
+// configureConnection sets up the WebSocket connection with a read deadline
+// and a handler for receiving pong messages from the client.
+//
+// The function does the following:
+//  1. Sets an initial read deadline to ensure the server doesn't wait indefinitely
+//     for a pong message from the client. If no message is received within the
+//     specified `pongWait` duration, the connection will be closed.
+//  2. Establishes a Pong handler that resets the read deadline every time a pong
+//     message is received from the client, allowing the server to continue waiting
+//     for further pong messages within the new deadline.
+func (c *Controller) configureConnection() error {
+	// Set the initial read deadline for the first pong message
```

Comment:
Can we also set a close handler here?

Reply:
I guess that we should think about`close handler` as part of our error handling subtask. The default `close handler `sends a close message back to the peer.